### ISSUE TITLE
The version of `protoc-gen-swift` is mismatched between the macOS and Linux versions of the latest release (`1.23.0`)

### ISSUE THREAD

**Issue by ezzatron** [grpc/grpc-swift#2052]
### Describe the bug

When downloading archives from the [`1.23.0` GitHub release](https://github.com/grpc/grpc-swift/releases/tag/1.23.0), the included version of `protoc-gen-swift` differs between the macOS and Linux versions:

- The Linux version (https://github.com/grpc/grpc-swift/releases/download/1.23.0/protoc-grpc-swift-plugins-linux-x86_64-1.23.0.zip) contains `protoc-gen-swift 1.26.0`
- The macOS version (https://github.com/grpc/grpc-swift/releases/download/1.23.0/protoc-grpc-swift-plugins-1.23.0.zip) contains `protoc-gen-swift 1.25.1`

This is leading to us seeing different generated output under CI vs. when run locally.

### To reproduce

See this reproduction repo / workflow run: https://github.com/ezzatron/grpc-swift-binary-version-repro/actions/runs/10804792826

Alternatively, set up a GitHub repo with this workflow, and run it to check the version difference:

```yml
on:
  push:

jobs:
  linux:
    runs-on: ubuntu-latest
    steps:
      - run: |
          curl -sSL https://github.com/grpc/grpc-swift/releases/download/1.23.0/protoc-grpc-swift-plugins-linux-x86_64-1.23.0.zip -o protoc-grpc-swift-plugins.zip
          unzip protoc-grpc-swift-plugins.zip
          ./protoc-gen-swift --version
          ./protoc-gen-grpc-swift --version

  macos:
    runs-on: macos-latest
    steps:
      - run: |
          curl -sSL https://github.com/grpc/grpc-swift/releases/download/1.23.0/protoc-grpc-swift-plugins-1.23.0.zip -o protoc-grpc-swift-plugins.zip
          unzip protoc-grpc-swift-plugins.zip
          bin/protoc-gen-swift --version
          bin/protoc-gen-grpc-swift --version
```

### Expected behaviour

The version of `protoc-gen-swift` should be the same across all architectures for a given release.

### Additional information

_N/A_

**Comment by glbrntt** [comment]
Thanks for filing this, it should be resolved by #2053

**Comment by ezzatron** [comment]
@glbrntt I tried to verify that this was fixed in `1.24.0`, but it seems like the normal archives are completely missing from the GitHub release: https://github.com/grpc/grpc-swift/releases/tag/1.24.0

I also tried with `1.23.1`, but ran into this error with the Linux version:

```
./protoc-gen-swift: symbol lookup error: ./protoc-gen-swift: undefined symbol: $s10Foundation15ContiguousBytesMp
```

(see https://github.com/ezzatron/grpc-swift-binary-version-repro/actions/runs/11228796504/job/31213284514#step:2:11)

**Comment by glbrntt** [comment]
Apologies, 1.24.0 was incorrectly tagged. Try 1.24.1 https://github.com/grpc/grpc-swift/releases/tag/1.24.1

--- 

This comes from `protoc-gen-swift` which is from the SwiftProtobuf project and is unrelated:

```
./protoc-gen-swift: symbol lookup error: ./protoc-gen-swift: undefined symbol: $s10Foundation15ContiguousBytesMp
```

`$s10Foundation15ContiguousBytesMp` demangled is "protocol descriptor for Foundation.ContiguousBytes" which is a bit surprising. I'd suggest try to reproduce that issue in isolation and opening an issue against the SwiftProtobuf repo.

**Comment by ezzatron** [comment]
Thanks! Looks to be resolved in `1.24.1`:
- macOS: https://github.com/ezzatron/grpc-swift-binary-version-repro/actions/runs/11230270644/job/31217301725#step:2:13
- Linux: https://github.com/ezzatron/grpc-swift-binary-version-repro/actions/runs/11230270644/job/31217301425#step:2:11
